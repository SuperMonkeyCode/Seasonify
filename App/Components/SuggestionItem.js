import React from 'react';
import {
  ListItem,
  Left,
  Body,
  Right,
  Thumbnail,
  Text,
  Icon,
  Button,
} from 'native-base';

const SuggestionItem = ({
  suggestionKey,
  onPress,
  posterURL,
  title,
  subtitle,
}) => {
  const onPressSuggestion = () => {
    onPress(suggestionKey);
  };
  return (
    <ListItem onPress={onPressSuggestion} avatar>
      <Left>
        {posterURL && <Thumbnail square source={{ uri: posterURL }} />}
        {!posterURL &&
          <Button transparent style={{ width: 57 }} /> /* FIXME: too tricky */}
      </Left>
      <Body>
        <Text>{title}</Text>
        <Text note>{subtitle}</Text>
      </Body>
      <Right>
        <Icon name="arrow-forward" />
      </Right>
    </ListItem>
  );
};

export default SuggestionItem;
