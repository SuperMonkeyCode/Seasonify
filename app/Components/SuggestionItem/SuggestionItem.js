/* @flow */
import {
  View,
  ListItem,
  Left,
  Body,
  Right,
  Thumbnail,
  Text,
  Icon,
  Button,
} from 'native-base';
import { compose, pure, withHandlers } from 'recompose';
import React from 'react';

type Props = {
  /* parent */
  suggestionKey: number,
  onPress: Function,
  poster?: string,
  title: string,
  subtitle?: string,
  alreadyAdded?: boolean,
  /* connect */
  /* state */
  /* handlers */
  handlePressSuggestion: Function,
};

const enhance = compose(
  pure,
  withHandlers({
    handlePressSuggestion: ({ alreadyAdded, suggestionKey, onPress }: Props) => () => {
      if (!alreadyAdded) onPress(suggestionKey);
    },
  }),
);

function SuggestionItem({
  poster,
  title,
  subtitle,
  alreadyAdded = false,
  handlePressSuggestion,
}: Props) {
  return (
    <View>
      <ListItem onPress={handlePressSuggestion} avatar>
        <Left>
          {poster && <Thumbnail square source={{ uri: poster }} />}
          {!poster &&
            <Button transparent style={{ width: 57 }} /> /* FIXME: too tricky */}
        </Left>
        <Body>
          <Text>{title}</Text>
          <Text note>{subtitle}</Text>
        </Body>
        {!alreadyAdded &&
          <Right>
            <Icon name="add-circle" />
          </Right>}
      </ListItem>
    </View>
  );
}

export default enhance(SuggestionItem);
