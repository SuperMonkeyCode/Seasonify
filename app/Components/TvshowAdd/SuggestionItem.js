/* @flow */
import {
  Body,
  Button,
  Icon,
  Left,
  ListItem,
  Right,
  Text,
  Thumbnail,
  View,
} from 'native-base';
import { Keyboard } from 'react-native';
import { compose, pure, withHandlers } from 'recompose';
import React from 'react';

type Props = {
  /* parent */
  navigation: Object,
  suggestionKey: number,
  alreadyAddedId?: string,
  onPress: Function,
  poster?: string,
  title: string,
  subtitle?: string,
  /* connect */
  /* HOC */
  handlePressSuggestion: Function,
};

const enhance = compose(
  withHandlers({
    handlePressSuggestion: ({
      navigation,
      suggestionKey,
      alreadyAddedId,
      onPress,
    }: Props) => () => {
      if (alreadyAddedId) {
        Keyboard.dismiss();
        navigation.navigate('TvshowDetailsPage', { tvshowId: alreadyAddedId });
      } else {
        onPress(suggestionKey);
      }
    },
  }),
  pure,
);

function SuggestionItem({
  poster,
  title,
  subtitle,
  alreadyAddedId,
  handlePressSuggestion,
}: Props) {
  return (
    <View>
      <ListItem avatar onPress={handlePressSuggestion}>
        <Left>
          {poster && <Thumbnail square source={{ uri: poster }} />}
          {!poster &&
            <Button transparent disabled style={{ width: 57 }} /> /* FIXME: too tricky */}
        </Left>
        <Body>
          <Text>
            {title}
          </Text>
          <Text note>
            {subtitle}
          </Text>
        </Body>
        {!alreadyAddedId &&
          <Right>
            <Icon name="add-circle" />
          </Right>}
      </ListItem>
    </View>
  );
}

export default enhance(SuggestionItem);
