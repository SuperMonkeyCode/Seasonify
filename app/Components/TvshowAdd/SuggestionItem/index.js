/* @flow */
export { default } from './SuggestionItem';
