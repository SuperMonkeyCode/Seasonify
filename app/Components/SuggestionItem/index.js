/* @flow */
import { connect } from 'react-redux';

import { tvshowSelectors } from 'Store/tvshowStore';

import SuggestionItem from './SuggestionItem';

export default connect(
  state => ({
    codes: tvshowSelectors.getCodes(state),
  }),
  {},
)(SuggestionItem);
