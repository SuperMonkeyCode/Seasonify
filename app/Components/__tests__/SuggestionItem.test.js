import React from 'react';
import SuggestionItem from '../SuggestionItem';

function setup(specificProps = {}) {
  const props = {
    suggestionKey: 0,
    poster: 'http://url//poster.jpg',
    title: 'Deadwood',
    subtitle: '2004',
    alreadyAdded: false,
    onPress: jest.fn(),
    ...specificProps,
  };
  const component = shallow(<SuggestionItem {...props} />);
  return {
    component,
    props,
  };
}

describe('Rendering', () => {
  it('should render', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });

  it('should render, when no poster', () => {
    const { component } = setup({ poster: undefined });
    expect(component).toMatchSnapshot();
  });

  it('should render, when already added', () => {
    const { component } = setup({ alreadyAdded: true });
    expect(component).toMatchSnapshot();
  });
});

describe('Events', () => {
  it('should call onPress, when pressing the list item (not already added)', () => {
    const { component, props } = setup({ alreadyAdded: false });
    component.find('Styled(ListItem)').simulate('press');
    expect(props.onPress).toBeCalled();
  });

  it("should'nt call onPress, when pressing the list item (already added)", () => {
    const { component, props } = setup({ alreadyAdded: true });
    component.find('Styled(ListItem)').simulate('press');
    expect(props.onPress).not.toBeCalled();
  });
});
