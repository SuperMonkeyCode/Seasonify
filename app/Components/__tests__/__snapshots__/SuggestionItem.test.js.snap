// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should render 1`] = `
<Styled(ViewNB)>
  <Styled(ListItem)
    avatar={true}
    onPress={[Function]}
  >
    <Styled(Left)>
      <Styled(Thumbnail)
        source={
          Object {
            "uri": "http://url//poster.jpg",
          }
        }
        square={true}
      />
    </Styled(Left)>
    <Styled(Body)>
      <Styled(Text)>
        Deadwood
      </Styled(Text)>
      <Styled(Text)
        note={true}
      >
        2004
      </Styled(Text)>
    </Styled(Body)>
    <Styled(Right)>
      <Styled(Icon)
        name="add-circle"
      />
    </Styled(Right)>
  </Styled(ListItem)>
</Styled(ViewNB)>
`;

exports[`Rendering should render, when already added 1`] = `
<Styled(ViewNB)>
  <Styled(ListItem)
    avatar={true}
    onPress={[Function]}
  >
    <Styled(Left)>
      <Styled(Thumbnail)
        source={
          Object {
            "uri": "http://url//poster.jpg",
          }
        }
        square={true}
      />
    </Styled(Left)>
    <Styled(Body)>
      <Styled(Text)>
        Deadwood
      </Styled(Text)>
      <Styled(Text)
        note={true}
      >
        2004
      </Styled(Text)>
    </Styled(Body)>
  </Styled(ListItem)>
</Styled(ViewNB)>
`;

exports[`Rendering should render, when no poster 1`] = `
<Styled(ViewNB)>
  <Styled(ListItem)
    avatar={true}
    onPress={[Function]}
  >
    <Styled(Left)>
      <Styled(Button)
        style={
          Object {
            "width": 57,
          }
        }
        transparent={true}
      />
    </Styled(Left)>
    <Styled(Body)>
      <Styled(Text)>
        Deadwood
      </Styled(Text)>
      <Styled(Text)
        note={true}
      >
        2004
      </Styled(Text)>
    </Styled(Body)>
    <Styled(Right)>
      <Styled(Icon)
        name="add-circle"
      />
    </Styled(Right)>
  </Styled(ListItem)>
</Styled(ViewNB)>
`;
